### PR TITLE
Pkg 4937 pytest 7.4.4

### PR DESCRIPTION
**Pull request by Jrice1317**
pytest 7.4.4

**Destination channel:** {defaults}

### Links

- [PKG-4937](https://anaconda.atlassian.net/browse/PKG-4937) 
- [Upstream repository](https://github.com/pytest-dev/pytest/tree/7.4.4)
- [Upstream changelog/diff](https://docs.pytest.org/en/stable/changelog.html)

### Explanation of changes:

- Although the latest is version 8.2.2, updating to 7.4.4 for upcoming installer in case of breaking changes in major update

[PKG-4937]: https://anaconda.atlassian.net/browse/PKG-4937?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -43,16 +43,16 @@ test:
...
-    #- argcomplete
-    #- attrs >=19.2.0
-    #- hypothesis >=3.56
-    #- mock
-    #- nose
-    #- pygments >2.7.2
-    #- requests
-    #- setuptools
-    #- xmlschema
+    - argcomplete
```
Comment (danpetry) [inline recipe/meta.yaml:46]: why have these been uncommented, but the test commands haven't been changed?

Comment (skupr-anaconda) [reply recipe/meta.yaml:46]: They can be only required for [development](https://github.com/pytest-dev/pytest/commit/86945f9a1f7cae10a9ac9ccf4b41ce9ddfabe14c) not for testing

**Review by ViridianMelody — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (danpetry) [reply recipe/meta.yaml:46]: ok but dev requirements aren't needed in the context of building a package as far as I understand? are they used by the tests at all? if not, they should be deleted. If so, the tests should be run

Comment (Jrice1317) [reply recipe/meta.yaml]: They were not listed as development in version 7.4.4 (See here), but I have removed them and updated the test section.

**Review by danpetry — APPROVED** [review]

**Pull request merged**